test/system/src/test/java/io/pravega/test/system/ReadWriteAndScaleWithFailoverTest.java

```diff
@@ -0,0 +1,488 @@
+
+            //close all the writers
+            log.info("Closing writers");
+            writerList.forEach(writer -> writer.close());
+            //close all readers
+            log.info("Closing readers");
+            readerList.forEach(reader -> reader.close());
+            //delete readergroup
+            log.info("Deleting readergroup {}", readerGroupName);
+            readerGroupManager.deleteReaderGroup(readerGroupName);
+        }
+        //seal the stream
+        CompletableFuture<Boolean> sealStreamStatus = controller.sealStream(scope, STREAM_NAME);
+        log.info("Sealing stream {}", STREAM_NAME);
+        assertTrue(sealStreamStatus.get());
+        //delete the stream
+        CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(scope, STREAM_NAME);
+        log.info("Deleting stream {}", STREAM_NAME);
+        assertTrue(deleteStreamStatus.get());
+
+        //delete the scope
+        CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(scope);
+        log.info("Deleting scope {}", scope);
+        assertTrue(deleteScopeStatus.get());
+        log.info("Test {} succeeds ", "ReadWriteAndScaleWithFailover");
+
+        }
+
+        private void waitForScaling() {
+
+            CompletableFuture<Void> testResult = Retry.withExpBackoff(10, 10, 40, ofSeconds(10).toMillis())
+                    .retryingOn(AbstractScaleTests.ScaleOperationNotDoneException.class)
+                    .throwingOn(RuntimeException.class)
+                    .run(() -> controller.getCurrentSegments(scope, STREAM_NAME)
+                            .thenAccept(x -> {
+                                int currentNumOfSegments = x.getSegments().size();
+                                if (currentNumOfSegments == 2) {
+                                    log.info("The current number of segments is equal to 2, ScaleOperation did not happen");
+                                    //Scaling operation did not happen, retry operation.
+                                    throw new AbstractScaleTests.ScaleOperationNotDoneException();
+                                } else if (currentNumOfSegments > 2) {
+                                    //scale operation successful.
+                                    log.info("Current Number of segments is {}", currentNumOfSegments);
+                                    stopWriteFlag.set(true);
+                                } else {
+                                    Assert.fail("Current number of Segments reduced to less than 2. Failure of test");
+                                }
+                            }));
+        }
+
+    private void performFailoverTest() throws InterruptedException {
+
+        long currentWriteCount1;
+        long currentReadCount1;
+
+        log.info("Test with 2 controller, SSS instances running and without a failover scenario");
+
+        currentWriteCount1 = eventWriteCount.get();
```

Comment:
these two, sleapTime and currentWirteCount don't need to separate their declaration from their initialization. It just makes the code longer.